main_crnn.py

```diff
@@ -0,0 +1,241 @@
+        yield data[i:i+batch_size]
+
+
+def create_dummy_data():
+    """Create small cardiac data based on patches for demo.
+
+    Note that in practice, at test time the method will need to be applied to
+    the whole volume. In addition, one would need more data to prevent
+    overfitting.
+
+    """
+    data = loadmat(join(project_root, './data/cardiac.mat'))['seq']
+    nx, ny, nt = data.shape
+    ny_red = 8
+    sl = ny//ny_red
+    data_t = np.transpose(data, (2, 0, 1))
+    
+    # Synthesize data by extracting patches
+    train = np.array([data_t[..., i:i+sl] for i in np.random.randint(0, sl*3, 20)])
+    validate = np.array([data_t[..., i:i+sl] for i in (sl*4, sl*5)])
+    test = np.array([data_t[..., i:i+sl] for i in (sl*6, sl*7)])
+
+    return train, validate, test
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--num_epoch', metavar='int', nargs=1, default=['10'],
+                        help='number of epochs')
+    parser.add_argument('--batch_size', metavar='int', nargs=1, default=['1'],
+                        help='batch size')
+    parser.add_argument('--lr', metavar='float', nargs=1,
+                        default=['0.001'], help='initial learning rate')
+    parser.add_argument('--acceleration_factor', metavar='float', nargs=1,
+                        default=['4.0'],
+                        help='Acceleration factor for k-space sampling')
+    parser.add_argument('--debug', action='store_true', help='debug mode')
+    parser.add_argument('--savefig', action='store_true',
+                        help='Save output images and masks')
+
+    args = parser.parse_args()
+    cuda = True if torch.cuda.is_available() else False
+    Tensor = torch.cuda.FloatTensor if cuda else torch.Tensor
+
+    # Project config
+    model_name = 'crnn_mri'
+    acc = float(args.acceleration_factor[0])  # undersampling rate
+    num_epoch = int(args.num_epoch[0])
+    batch_size = int(args.batch_size[0])
+    Nx, Ny, Nt = 256, 256, 30
+    Ny_red = 8
+    save_fig = args.savefig
+    save_every = 5
+
+    # Configure directory info
+    project_root = '.'
+    save_dir = join(project_root, 'models/%s' % model_name)
+    if not os.path.isdir(save_dir):
+        os.makedirs(save_dir)
+
+    # Create dataset
+    train, validate, test = create_dummy_data()
+
+    # Test creating mask and compute the acceleration rate
+    dummy_mask = cs.cartesian_mask((10, Nx, Ny//Ny_red), acc, sample_n=8)
+    sample_und_factor = cs.undersampling_rate(dummy_mask)
+    print('Undersampling Rate: {:.2f}'.format(sample_und_factor))
+
+    # Specify network
+    rec_net = CRNN_MRI()
+    rec_net = rec_net.cuda()
+    criterion = torch.nn.MSELoss()
+    criterion.cuda()
+    optimizer = optim.Adam(rec_net.parameters(), lr=float(args.lr[0]), betas=(0.5, 0.999))
+
+    # # build CRNN-MRI with pre-trained parameters
+    # rec_net.load_state_dict(torch.load('./models/pretrained/crnn_mri_d5_c5.pth'))
```

Comment:
I think you need to add this before `rec_net.cuda()`, unless you saved the model in `CudaTensor` format. Can you make sure your model is saved as `Tensor`? Otherwise people would need GPU to load it...